src/plugins/data/public/query/query_string/language_service/lib/query_result.tsx

```diff
@@ -22,8 +22,16 @@ export interface QueryStatus {
   status: ResultStatus;
   body?: {
     error?: {
+      error?: string;
+      message?: {
+        error?: {
+          reason?: string;
+          details: string;
+          type?: string;
+        };
+        status?: number;
+      };
       statusCode?: number;
```

Comment:
Not for this PR, mostly for my understanding: Is this the structure of the data passed from OS? Seems like a big code smell to me that we have 3 levels of error and 2 of statuses.